### PR TITLE
feat: add http_request_next method (large asset support)

### PR DESCRIPTION
**Pull request by ericswanson-dfinity**
Part of https://github.com/dfinity/dx-triage/issues/126

See also https://github.com/dfinity/sdk/pull/1512

### CODE REVIEW

File: ic-utils/src/interfaces/http_request.rs [hunk 1]
```diff
@@ -29,6 +29,18 @@ pub struct HttpResponse {
...
+#[derive(CandidType)]
+pub struct HttpGetChunkRequest {
+    pub url: String,
+    pub content_encoding: String,
+    pub index: Nat,
+}
+
+#[derive(CandidType, Deserialize)]
+pub struct HttpGetChunkResponse {
+    pub chunk: Vec<u8>,
```
Comment (p-shahi) [inline ic-utils/src/interfaces/http_request.rs:41]:
```suggestion
    #[serde(with = "serde_bytes")]
    pub chunk: Vec<u8>,
```

**Review by p-shahi — APPROVED** [review]

File: ic-utils/src/interfaces/http_request.rs [hunk 2]
```diff
@@ -29,6 +29,18 @@ pub struct HttpResponse {
     pub body: Vec<u8>,
 }
 
+#[derive(CandidType)]
+pub struct HttpGetChunkRequest {
```
Comment (p-shahi) [inline ic-utils/src/interfaces/http_request.rs:33]: can this be moved inside the body of `http_get_chunk` similar to https://github.com/dfinity/agent-rs/blob/next/ic-utils/src/interfaces/management_canister.rs#L453

**Review by hansl — CHANGES_REQUESTED** [review]
Need design. See dfinity/dx-triage#126 where I'll start the discussion.

File: ic-utils/src/canister.rs [hunk 3]
```diff
@@ -192,6 +193,25 @@ impl Argument {
         }
     }
 
+    /// Add an IDLValue Argument. If the current value of Argument is Raw, will set the
+    /// result to an error. If the current value is an error, will do nothing.
+    pub fn push_value_arg(&mut self, arg: IDLValue) {
```
Comment (hansl) [inline ic-utils/src/canister.rs:198]: `IDLValue` implements CandidType, so you should be able to use `push_idl_arg` with an `IDLValue`. Was there any issue with using that other function?

File: ic-utils/src/canister.rs [hunk 4]
```diff
@@ -258,6 +278,16 @@ impl<'agent, 'canister: 'agent, Interface> SyncCallBuilder<'agent, 'canister, In
         self
     }
 
+    /// Add an argument to the candid argument list. This requires Candid arguments, if
+    /// there is a raw argument set (using [with_arg_raw]), this will fail.
+    pub fn with_value_arg(
```
Comment (hansl) [inline ic-utils/src/canister.rs:283]: Same question as above.

**Review by hansl — CHANGES_REQUESTED** [review]

File: ic-utils/src/canister.rs (hunk 3 above)
Comment (ericswanson-dfinity) [reply ic-utils/src/canister.rs:198]: Yes, `IDLValue::ty()` calls unreachable() : https://github.com/dfinity/candid/blob/master/rust/candid/src/parser/value.rs#L326
called from here: https://github.com/dfinity/candid/blob/master/rust/candid/src/ser.rs#L28

**Review by hansl — CHANGES_REQUESTED** [review]
.

**Review by hansl — APPROVED** [review]
Okay LGTM, but this is an anti-pattern. We might have to deprecate these methods if we make IDLValue work.

**Comment by ericswanson-dfinity** [comment]
Entered https://github.com/dfinity/agent-rs/issues/132 to make `with_arg` work with `IDLValue`.

**Review by hansl — APPROVED** [review]

**Pull request merged**